Cesium3DTilesSelection/src/TilesetContentManager.cpp

```diff
@@ -708,9 +709,16 @@ TilesetContentManager::TilesetContentManager(
               // and create corresponding loader
               const auto rootIt = tilesetJson.FindMember("root");
               if (rootIt != tilesetJson.MemberEnd()) {
-                TilesetContentLoaderResult<TilesetContentLoader> result =
-                    TilesetJsonLoader::createLoader(pLogger, url, tilesetJson);
-                return asyncSystem.createResolvedFuture(std::move(result));
+                return TilesetJsonLoader::createLoader(
+                           asyncSystem,
+                           pAssetAccessor,
+                           pLogger,
+                           url,
+                           requestHeaders,
+                           tilesetJson)
+                    .thenImmediately(
+                        [](TilesetContentLoaderResult<TilesetContentLoader>&&
+                               result) { return std::move(result); });
```

Comment:
This `thenImmediately` isn't doing anything AFAICT. It can be removed completely.

Reply:
Unfortunately if I don't include this, I get these build errors:
```
3>C:\Dev\cesium-native\Cesium3DTilesSelection\src\TilesetContentManager.cpp(718,1): error C3487: 'CesiumAsync::Future<Cesium3DTilesSelection::TilesetContentLoaderResult<Cesium3DTilesSelection::TilesetJsonLoader>>': all return expressions must deduce to the same type: previously it was 'CesiumAsync::Future<Cesium3DTilesSelection::TilesetContentLoaderResult<Cesium3DTilesSelection::TilesetContentLoader>>'
3>C:\Dev\cesium-native\Cesium3DTilesSelection\src\TilesetContentManager.cpp(712,17): error C2440: 'return': cannot convert from 'CesiumAsync::Future<Cesium3DTilesSelection::TilesetContentLoaderResult<Cesium3DTilesSelection::TilesetJsonLoader>>' to 'CesiumAsync::Future<Cesium3DTilesSelection::TilesetContentLoaderResult<Cesium3DTilesSelection::TilesetContentLoader>>'
3>C:\Dev\cesium-native\Cesium3DTilesSelection\src\TilesetContentManager.cpp(712,17): message : No constructor could take the source type, or constructor overload resolution was ambiguous
```

I saw something similar being done for `LayerJsonTerrainLoader` several lines down, so I thought this was just a necessary workaround 😅

Reply:
Yikes it's relying an implicit conversion of `TilesetContentLoaderResult<TilesetJsonLoader>` to `TilesetContentLoaderResult<TilesetContentLoader>`. The implicit conversion happens when calling the `thenImmediately` lambda. That's really not obvious from the code!

Anyway, as you mentioned, it's done elsewhere. And it's not very obvious what we could do that would be better, so :shipit:.